Adam repositories controllers

aceralrighty opened the pull request:
### **Description**  
<!-- Provide a summary of the changes, explaining why they are needed and how they solve the problem. -->  

---

### **Frontend**  
- [ ] UI changes included  
- [ ] New Angular components/services added  
- [ ] Routing updates  
- [ ] Jest tests added/updated  
- [ ] Responsiveness tested  
- [ ] Accessibility considerations  

---

### **Backend**  
- [ ] New REST endpoints added  
- [ ] Existing endpoints modified  
- [ ] Security/OAuth updates  
- [ ] Database migrations (if any)  
- [ ] Unit/integration tests added/updated  
- [ ] API docs updated (if applicable)  

---

### **Testing**  
- [ ] Locally tested in dev environment  
- [ ] Angular unit tests passing (`npm run test`)  
- [ ] Spring Boot tests passing (`./mvnw test`)  
- [ ] Manual tests performed (describe below):  

**Test Cases:**  
1.  
2.  
3.  

---

### **Deployment Notes**  
- [ ] Requires database migration  
- [ ] Requires environment variable changes  
- [ ] Requires special deployment instructions (describe below)  

---

### **Additional Context/Screenshots**  
<!-- Add any other context, screenshots, or performance notes here -->  

---

### **Reviewer Checklist**  
- [ ] Code adheres to project style guidelines  
- [ ] No console errors  
- [ ] Code is self-explanatory with proper comments  
- [ ] No sensitive data/exposed credentials  
- [ ] Functionality works as expected  
- [ ] PR description is complete and clear

aceralrighty committed:
Merge branch 'main' of https://github.com/BauschBakeries/BauschBakeries
just pulling in some stuff from my models branch

aceralrighty committed:
Merge branch 'Adam_models' of https://github.com/BauschBakeries/BauschBakeries
just get me my stuff!!

aceralrighty committed:
:beginning of the service layer, i hope you like the naming conventions

aceralrighty committed:
:some more services

aceralrighty committed:
another service

JosefBausch commented:
the _I_ prefix should be reserved to interfaces for classes/methods. "I" denotes interface. Services should follow the pattern of "ModelNameService" or something similar, but always ending in service. I will be creating an interface directory here shortly, where we can base our backend methods off of.

aceralrighty committed:
:made those changes, masta

JosefBausch committed:
B
Merge branch 'main' of https://github.com/BauschBakeries/BauschBakeries into Adam_repositories_controllers

 merging main

aceralrighty committed:
:got the changes

JosefBausch committed:
Merge branch 'Adam_repositories_controllers' of https://github.com/BauschBakeries/BauschBakeries into Adam_repositories_controller

merged main in

aceralrighty committed:
fixed some naming issues

JosefBausch reviewed (changes requested):
Please go through and change the incorrect service names.
IServiceLayer needs to be changed to Services, I believe that makes the most sense.
Also, please create a generic interfaces directory, we will need that when we write controllers and internal methods for compute.

aceralrighty committed:
:added interface directory. renamed interface implementation

aceralrighty committed:
:nothing new

aceralrighty committed:
:i added another service layer just to start. let me know what you think of this approach. i was running into problems solely using the UsersRepository

aceralrighty committed:
:added another method
